src/main/java/com/hortonworks/spark/sql/hive/llap/HiveWarehouseSession.java

```diff
@@ -0,0 +1,36 @@
+package com.hortonworks.spark.sql.hive.llap;
+
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+
+public interface HiveWarehouseSession {
+
+    String HIVE_WAREHOUSE_CONNECTOR = "com.hortonworks.spark.sql.hive.llap.HiveWarehouseConnector";
+
+    Dataset<Row> executeQuery(String sql);
+    Dataset<Row> q(String sql);
+
+    Dataset<Row> execute(String sql);
+    Dataset<Row> exec(String sql);
```

Comment:
@EricWohlstadter, sorry for leaving a comment late like this but `exec` is a reserved keyword in Python. If you don't mind will just take this out in Python. Seems a short name for `execute`. Python side uses a trailing underscore, for example, `exec_` in this case but I think it's fine to not add it if it's just a short name for `execute`.

Reply:
+1. Could you open an issue for that, @HyukjinKwon ?

Reply:
@HyukjinKwon 

Ok, let's just take out "exec" all together. Please open an issue and I'll make the change for the Java side.